chain/chain/src/flat_storage_resharder.rs

```diff
@@ -631,6 +646,22 @@ impl FlatStorageResharder {
 
         let mut num_batches_done: usize = 0;
 
+        let status = self
+            .runtime
+            .store()
+            .flat_store()
+            .get_flat_storage_status(shard_uid)
+            .map_err(|e| Into::<StorageError>::into(e))?;
+        let FlatStorageStatus::Resharding(FlatStorageReshardingStatus::CatchingUp(
```

Comment:
What happens when the catchup is postponed and `mark_flat_storage_ready` is called? Won't this line fail because the status is not catchup anymore?

Reply:
It should be fine because `mark_flat_storage_ready()` is called on a [checkpoint](https://github.com/near/nearcore/blob/beb00f4aa1b344a196012b9357b02e21cf167aab/core/store/src/trie/state_snapshot.rs#L209) of the main RocksDB, so this code won't see those changes